homeassistant/components/sensor/temper.py

```diff
@@ -23,18 +34,28 @@ def setup_platform(hass, config, add_devices_callback, discovery_info=None):
 
     temp_unit = hass.config.units.temperature_unit
     name = config.get(CONF_NAME, DEVICE_DEFAULT_NAME)
```

Comment:
Because name is optional, I think it will now exist in `config` but have the value `None` if not given. That means that when omitting name in config, the value will be None and give an error. Instead, add a default value:

```python
vol.Optional(CONF_NAME, default=DEVICE_DEFAULT_NAME)
```